pageserver/src/tenant/storage_layer/inmemory_layer.rs

```diff
@@ -621,6 +624,26 @@ impl InMemoryLayer {
 
         let end_lsn = *self.end_lsn.get().unwrap();
 
+        // Sort the keys because delta layer writer expects them sorted.
+        //
+        // NOTE: this sort can take up significant time if the layer has millions of
+        //       keys. To speed up all the comparisons we convert the key to i128 and
```

Comment:
How much savings does this give us in terms of bytes? Also, have you verified that the conversion function preserves the order, so `k < j iff k.to_i128() < j.to_i128()` for all keys `k,j`.

Reply:
I moved the original code to this place and kept the original comments, and I do doubt whether it is necessary to do to_i128 for delta layers...